rules_128tech/pylint_plugins/mock_patch_autospec.py

```diff
@@ -0,0 +1,69 @@
+"""
+A plugin to detect and disallow the use of mock.patch or mocker.patch without
+setting autospec=True.
+
+Based on http://pylint.pycqa.org/en/latest/how_tos/custom_checkers.html#write-a-checker
+"""
+
+import astroid
+import astroid.node_classes
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+
+
+def register(linter):
+    linter.register_checker(MockPatchAutospecChecker(linter))
+
+
+class MockPatchAutospecChecker(BaseChecker):
+    """A pylint checker to detect use of mock.patch without autospec=True"""
+
+    __implements__ = IAstroidChecker
+
+    name = "mock-patch-autospec"
+    priority = -1
+    msgs = {
+        "W1280": (
+            "Uses mock.patch without autospec=True.",
+            name,
+            (
+                "All mock patches should use autospec=True to avoid allowing calls "
+                "to functions that don't exist in production code"
+            ),
+        ),
+    }
+
+    def visit_call(self, call: astroid.node_classes.Call) -> None:
+        if not isinstance(call.func, astroid.node_classes.Attribute):
+            # Not a method call, not applicable
+            return
+
+        if call.func.attrname != "patch":
+            # Some other method, not applicable
+            return
+
+        if not isinstance(call.func.expr, astroid.node_classes.Name):
+            # A method call on a complex expression, literal, or something
+            return
+
+        if call.func.expr.name not in ("mock", "mocker"):
+            # Call to .patch() on some other object, I guess...
+            return
+
+        if call.keywords:
+            for keyword in call.keywords:
```

Comment:
if you go the route of just making sure that the argument is set then the implementation could be simplified
https://github.com/PyCQA/pylint/blob/1736d3a9f0768f62b3d07f2d93a010eb92242633/pylint/checkers/stdlib.py#L292

Reply:
The logic got a bit more complicated so I kept it mostly the same, but it could be done either way at this point. Actually... I might be able to do something fancy with set intersection here